lib/native/aws_iot.ts

```diff
@@ -445,7 +410,7 @@ export class AwsIotMqttConnectionConfigBuilder {
 
         // Check to see if a custom authorizer is being used but not through the builder
         if (this.is_using_custom_authorizer == false) {
-            if (this.params.username != "" && this.params.username != null && this.params.username != undefined) {
+            if (iot_shared.is_string_and_not_empty(this.params.username) && this.params.username) {
```

Comment:
Do we need the this.params.username check?  Seems like the helper function includes that.